Library/Homebrew/livecheck/strategy/xorg.rb

```diff
@@ -84,15 +82,12 @@ def self.find_versions(url, regex)
 
           regex ||= /href=.*?#{Regexp.escape(module_name)}[._-]v?(\d+(?:\.\d+)+)\.t/i
 
-          match_data = { matches: {}, regex: regex, url: page_url }
-
-          # Cache responses to avoid unnecessary duplicate fetches
-          @page_data[page_url] = URI.parse(page_url).open.read unless @page_data.key?(page_url)
+          # Use the cached page content to avoid duplicate fetches
+          cached_content = @page_data[page_url]
```

Comment:
Where is this cached content coming from?

Reply:
`Xorg` formulae/casks basically end up checking a small number of directory listing pages that contain a bunch of different software. Rather than checking the same pages over and over, the strategy saves the HTML in `@page_data`, using the page URL as the key. This way, we only fetch the page once. It also makes subsequent checks [using the same page] very fast, so running through homebrew-xorg is faster as a result.

This naive approach is fine within this limited context but I have something more sophisticated in the works for future livecheck-wide caching. This `Xorg` logic is just something I threw together while refactoring the strategy as part of migrating it from homebrew-xorg (when some X.org formulae were being migrated to homebrew/core).